crates/dprint/src/utils/glob/glob_matcher.rs

```diff
@@ -77,12 +77,15 @@ impl GlobMatcher {
         exclude_matcher,
       } => {
         let path = path.as_ref();
-        let path = if path.is_absolute() {
+        let path = if path.is_absolute() && path.starts_with(&base_dir) {
           Cow::Borrowed(path.strip_prefix(&base_dir).unwrap())
         } else {
           Cow::Owned(base_dir.join(path))
         };
-        matches!(include_matcher.matched(&path, false), Match::Whitelist(_)) && !matches!(exclude_matcher.matched(&path, false), Match::Whitelist(_))
+
+        path.starts_with(&base_dir) 
+          && matches!(include_matcher.matched(&path, false), Match::Whitelist(_)) 
+          && !matches!(exclude_matcher.matched(&path, false), Match::Whitelist(_))
```

Comment:
I tried to come up with something smart with the path but failed. I am a bit too new to rust to figure it out.